packages/zwave-js/src/lib/commandclass/HumidityControlModeCC.ts

```diff
@@ -0,0 +1,383 @@
+import type { ValueID } from "@zwave-js/core";
+import {
+	CommandClasses,
+	enumValuesToMetadataStates,
+	Maybe,
+	MessageOrCCLogEntry,
+	MessageRecord,
+	parseBitMask,
+	validatePayload,
+	ValueMetadata,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import { getEnumMemberName, pick } from "@zwave-js/shared";
+import type { Driver } from "../driver/Driver";
+import { MessagePriority } from "../message/Constants";
+import {
+	CCAPI,
+	PollValueImplementation,
+	POLL_VALUE,
+	SetValueImplementation,
+	SET_VALUE,
+	throwUnsupportedProperty,
+	throwWrongValueType,
+} from "./API";
+import {
+	API,
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+
+// All the supported commands
+export enum HumidityControlModeCommand {
+	Set = 0x01,
+	Get = 0x02,
+	Report = 0x03,
+	SupportedGet = 0x04,
+	SupportedReport = 0x05,
+}
+
+export enum HumidityControlMode {
+	"Off" = 0x00,
+	"Humidify" = 0x01,
+	"De-humidify" = 0x02,
+	"Auto" = 0x03,
+}
+
+@API(CommandClasses["Humidity Control Mode"])
+export class HumidityControlModeCCAPI extends CCAPI {
+	public supportsCommand(cmd: HumidityControlModeCommand): Maybe<boolean> {
+		switch (cmd) {
+			case HumidityControlModeCommand.Get:
+			case HumidityControlModeCommand.SupportedGet:
+				return this.isSinglecast();
+			case HumidityControlModeCommand.Set:
+				return true; // This is mandatory
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property },
+		value,
+	): Promise<void> => {
+		const valueDB = this.endpoint.getNodeUnsafe()!.valueDB;
+		if (property === "mode") {
+			if (typeof value !== "number") {
+				throwWrongValueType(
+					this.ccId,
+					property,
+					"number",
+					typeof value,
+				);
+			}
+		} else {
+			throwUnsupportedProperty(this.ccId, property);
+		}
+
+		if (this.isSinglecast()) {
+			this.schedulePoll({ property });
+		}
+	};
+
+	protected [POLL_VALUE]: PollValueImplementation = async ({
+		property,
+	}): Promise<unknown> => {
+		switch (property) {
+			case "mode":
+				return (await this.get())?.[property];
+
+			default:
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
I changed this one to return `response?.mode` rather than use `pick()`, similar to the Operating State. With quick search through the code I see that most methods that `return pick()` also disable the warning. What other fix did you have in mind? Perhaps adding an interface with work, I can do that for the Setpoint `get()` api?